src/Draco.Compiler/Api/CodeFixes/CodeFixProvider.cs

```diff
@@ -0,0 +1,17 @@
+using System.Collections.Immutable;
+using Draco.Compiler.Api.Diagnostics;
+
+namespace Draco.Compiler.Api.CodeFixes;
+
+/// <summary>
+/// Base class for providing <see cref="CodeFix"/>es.
+/// </summary>
+public abstract class CodeFixProvider
+{
+    /// <summary>
+    /// Gets all <see cref="CodeFix"/>es from this <see cref="CodeFixProvider"/>.
+    /// </summary>
+    /// <param name="diagnostics">Current <see cref="Diagnostic"/>s.</param>
+    /// <returns>All <see cref="CodeFix"/>es from this <see cref="CodeFixProvider"/>.</returns>
+    public abstract ImmutableArray<CodeFix> GetCodeFixes(ImmutableArray<Diagnostic> diagnostics);
+}
```

Comment:
Bad API, the user will hate to implement this.
A code fix provider should be able to tell what diag codes it can fix. Then the service will match up providers with diag codes. The `GetCodeFixes` method should receive the single diagnostic it should handle, the syntax tree and semantic model